uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -910,6 +910,219 @@ public function testManagerCanCancelOrRejectLeaveRequestEvenIfBalanceIsGreaterTh
     }
   }
 
+  public function testManagerCanNotApproveLeaveRequestIfBalanceChangeIsMoreThanEntitlementBalanceWhenAllowOveruseFalse() {
+    $manager = ContactFabricator::fabricate();
+    $staff = ContactFabricator::fabricate();
+    $periodStartDate = CRM_Utils_Date::processDate('2016-01-01');
+    $periodEndDate = CRM_Utils_Date::processDate('2016-12-31');
+    $requestDate = CRM_Utils_Date::processDate('2016-06-14');
+    $requestDateType = $this->leaveRequestDayTypes['all_day']['value'];
+
+    $this->registerCurrentLoggedInContactInSession($manager['id']);
+    $this->setContactAsLeaveApproverOf($manager, $staff);
+
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => $periodStartDate,
+      'end_date'   => $periodEndDate,
+    ]);
+
+    $absenceType = AbsenceTypeFabricator::fabricate(['allow_overuse' => 0]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $absenceType->id,
+      'contact_id' => $staff['id'],
+      'period_id' => $absencePeriod->id
+    ]);
+
+    // Current balance must be less than the balance change
+    $entitlementBalance = 0;
+    $this->createLeaveBalanceChange($periodEntitlement->id, $entitlementBalance);
+    $periodStartDate = $absencePeriod->start_date;
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $periodEntitlement->contact_id],
+      ['period_start_date' => $periodStartDate]
+    );
+
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => 1]);
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $staff['id'],
+      'pattern_id' => $workPattern->id
+    ]);
+
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id', 'validate'));
+
+    // Balance change is 1 since this is a single day leave request
+    $leaveRequestData = [
+      'type_id' => $absenceType->id,
+      'contact_id' => $periodEntitlement->contact_id,
+      'status_id' => $leaveRequestStatuses['awaiting_approval'],
+      'from_date' => $requestDate,
+      'from_date_type' => $requestDateType,
+      'to_date' => $requestDate,
+      'to_date_type' => $requestDateType,
+      'request_type' => LeaveRequest::REQUEST_TYPE_LEAVE
+    ];
+
+    $this->setExpectedException(
+      'CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException',
+      'There are only '. $entitlementBalance .' days leave available. This request cannot be made or approved'
+    );
+
+    // Create new request by staff with Awaiting Approval status
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation($leaveRequestData, true);
```

Comment:
It would be better to move this statement before the `setExpectedException()` one, just to make sure that the exception will be thrown by LeaveRequest::create().

This applies to the other tests